src/main/java/snowy/TaskList.java

```diff
@@ -0,0 +1,243 @@
+package snowy;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of Tasks with the methods the add, remove and modify the tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Creates a TaskList with tasks strings that are previously saved in a file.
+     * @param lines the ArrayList with each element containing the task as a string.
+     */
+    public TaskList(ArrayList<String> lines) {
+        for (String line : lines) {
+            initializeTask(line);
+        }
+    }
+
+    /**
+     * Creates an empty TaskList.
+     */
+    public TaskList(){
+
+    }
+
+    private void initializeTask(String description) {
+        String status;
+        String name;
+
+        String[] dataArray = description.split("[|]");
+        String type = dataArray[0];
+        switch (type) {
+        case "T":
+            status = dataArray[1];
+            name = dataArray[2];
+            Task newToDo = new ToDo(name);
+            tasks.add(newToDo);
+            if (status.equals("1")) {
+                newToDo.markComplete();
+            }
+            break;
+
+        case "D":
+            status = dataArray[1];
+            name = dataArray[2];
+            String dueDate = dataArray[3];
+            try {
+                Task newDateline = new Deadline(name, dueDate);
+                tasks.add(newDateline);
+                if (status.equals("1")) {
+                    newDateline.markComplete();
```

Comment:
Should `newDateline` be `newDeadline` instead? I think it might be a typo - do check!